src/main/java/org/generationcp/middleware/api/study/StudyEntryObservationServiceImpl.java

```diff
@@ -32,12 +33,18 @@ public Integer createObservation(final StockPropertyData stockPropertyData) {
 	}
 
 	@Override
-	public Integer updateObservation(final StockPropertyData stockPropertyData) {
-		final StockProperty stockProperty =
-			this.daoFactory.getStockPropertyDao().getByStockIdAndTypeId(stockPropertyData.getStockId(), stockPropertyData.getVariableId())
-				.orElseThrow(() -> new MiddlewareException(
-					"Stock property: " + stockPropertyData.getStockId() + " with type: " + stockPropertyData.getVariableId()
-						+ " not found."));
+	public Integer updateObservation(final StockPropertyData stockPropertyData, final boolean allowCreate) {
+		final Optional<StockProperty> stockPropertyOptional = this.daoFactory.getStockPropertyDao().getByStockIdAndTypeId(
+			stockPropertyData.getStockId(), stockPropertyData.getVariableId());
+
+		if (!stockPropertyOptional.isPresent() && allowCreate) {
```

Comment:
Move the logic that checks for allowCreate and creation of observation outside of this method, so that it's clear that this is just for updating of observation.

Reply:
created new method: createOrUpdateObservation